Sources/DatadogSDKTesting/Environment/Environment.swift

```diff
@@ -0,0 +1,385 @@
+            }
+        }
+        if env.has("DD_GIT_COMMIT_SHA") {
+            if let sha = env["DD_GIT_COMMIT_SHA", String.self] {
+                if !sha.isHexNumber {
+                    log.print("DD_GIT_COMMIT_SHA environment variable was configured with a non-hexadecimal string")
+                } else if sha.count != 40 {
+                    log.print("DD_GIT_COMMIT_SHA environment variable was configured with a value shorter than 40 character")
+                }
+            } else {
+                log.print("DD_GIT_COMMIT_SHA environment variable was configured with an empty value")
+            }
+        }
+    }
+    
+    private static func gitOverrides(env: EnvironmentReader) -> Git {
+        let (branch, isTag) = Git.normalize(branchOrTag: env["DD_GIT_BRANCH"])
+        return .init(
+            repositoryURL: env["DD_GIT_REPOSITORY_URL"],
+            branch: isTag ? nil : branch,
+            tag: isTag ? branch : env["DD_GIT_TAG"],
+            commitSHA: env["DD_GIT_COMMIT_SHA"],
+            commitMessage: env["DD_GIT_COMMIT_MESSAGE"],
+            authorName: env["DD_GIT_COMMIT_AUTHOR_NAME"],
+            authorEmail: env["DD_GIT_COMMIT_AUTHOR_EMAIL"],
+            authorDate: env["DD_GIT_COMMIT_AUTHOR_DATE"],
+            committerName: env["DD_GIT_COMMIT_COMMITTER_NAME"],
+            committerEmail: env["DD_GIT_COMMIT_COMMITTER_EMAIL"],
+            committerDate: env["DD_GIT_COMMIT_COMMITTER_DATE"]
+        )
+    }
+    
+    static func gitInfoAt(startingPath: String) -> GitInfo? {
+        var rootFolder = NSString(string: URL(fileURLWithPath: startingPath).path)
+        while !FileManager.default.fileExists(atPath: rootFolder.appendingPathComponent(".git")) {
+            if rootFolder.isEqual(to: rootFolder.deletingLastPathComponent) {
+                // We reached to the top
+                // Log.print("could not find .git folder at \(rootFolder)")
+                break
+            }
+            rootFolder = rootFolder.deletingLastPathComponent as NSString
+        }
+        let rootDirectory = URL(fileURLWithPath: rootFolder as String, isDirectory: true)
+        let gitInfo = try? GitInfo(gitFolder: rootDirectory.appendingPathComponent(".git"))
+        return gitInfo
+    }
+
+    
+    static var ciReaders: [CIEnvironmentReader] {
+        return [
+            TravisCIEnvironmentReader(), CircleCIEnvironmentReader(), JenkinsCIEnvironmentReader(),
+            GitlabCIEnvironmentReader(), AppveyorCIEnvironmentReader(), AzureCIEnvironmentReader(),
+            BitbucketCIEnvironmentReader(), GithubCIEnvironmentReader(), BuildkiteCIEnvironmentReader(),
+            BitriseCIEnvironmentReader(), XcodeCIEnvironmentReader(), TeamcityCIEnvironmentReader(),
+            CodefreshCIEnvironmentReader(), AwsCodePipelineCIEnvironmentReader(),
+            AwsCodeBuildCIEnvironmentReader(), BuddyCIEnvironmentReader()
+        ]
+    }
```

Comment:
I have something similar in .NET but I have also a Custom one for reading `DD_GIT` env vars

Reply:
I handled "DD_" variables as special case, they override values from the "CI" environment and can be combined with any CI.